File: src/integrationTest/java/com/doestheohavethisbook/HelloWorldTests.java
package com.doestheohavethisbook;

import org.junit.Test;

import kong.unirest.HttpResponse;
import kong.unirest.Unirest;

import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;

public class HelloWorldTests {
    @Test
    public void whenICallHello_thenReturnsHelloWorld() {
        HttpResponse<String> response = Unirest.get("http://localhost:8080/hello").asString();

        assertThat(response.getStatus(), is(200));
        assertThat(response.getBody(), is("Hello, World!"));
    }
}

File: src/integrationTest/java/com/doestheohavethisbook/HelloWorldTests.kt
package com.doestheohavethisbook

import junit.framework.TestCase.assertEquals
import org.junit.Test

import kong.unirest.HttpStatus
import kong.unirest.Unirest

internal class HelloWorldTests {
    @Test
    fun whenICallHello_thenReturnsHelloWorld() {
        val response = Unirest.get("http://localhost:8080/hello").asString()

        assertEquals(HttpStatus.OK, response.getStatus())
        assertEquals("Hello, World!", response.getBody())
    }
}
